src/games/strategy/engine/framework/ui/NewGameChooser.java

```diff
@@ -246,26 +246,26 @@ private synchronized static void synchronizedClear() {
   private void refreshGameList() {
     m_gameList.setEnabled(false);
     final NewGameChooserEntry selected = getSelected();
-    SwingUtilities.invokeLater(new Runnable() {
-      @Override
-      public void run() {
         try {
           refreshNewGameChooserModel();
           m_gameListModel = getNewGameChooserModel();
           m_gameList.setModel(m_gameListModel);
-          if (selected != null) {
-            final String name = selected.getGameData().getGameName();
-            final NewGameChooserEntry found = m_gameListModel.findByName(name);
-            if (name != null) {
-              m_gameList.setSelectedValue(found, true);
-            }
+          if(selected != null){
+            selectGame(selected.getGameData().getGameName());
           }
         } finally {
           m_gameList.setEnabled(true);
         }
+  }
+  
+  private void refreshGameListBackground(){
```

Comment:
My thoughts behind this were, that since the "refresh maps" button called SwingUtillities.runLater I thought the sense behind this method was that the gui won't have to wait to be drawn until all maps are reloaded...
But in order get this bug fixed there is a need of not having a runLaterMethod.
So i implemented a second method running the runLater Method for the button and one without runLater for the initial reload process...